cf/data/array/cfanetcdfarray.py

```diff
@@ -195,83 +238,149 @@ def __init__(
         if aggregated_data is None:
             aggregated_data = self.get_aggregated_data(copy=False)
 
-        return (
-            self.__class__.__name__,
-            abspath(self.get_filename()),
-            self.get_ncvar(),
-            self.get_group(),
-            aggregated_data,
-        )
+        return out + (aggregated_data,)
 
     def __getitem__(self, indices):
         """x.__getitem__(indices) <==> x[indices]"""
         return NotImplemented  # pragma: no cover
 
-    def _set_fragment(self, var, frag_loc, aggregated_data, cfa_filename):
-        """Create a new key/value pair in the *aggregated_data*
-        dictionary.
+    def get_aggregated_data(self, copy=True):
+        """Get the aggregation data dictionary.
 
-        The *aggregated_data* dictionary contains the definitions of
-        the fragments and the instructions on how to aggregate them,
-        and is updated in-place.
+        The aggregation data dictionary contains the definitions of
+        the fragments and the instructions on how to aggregate them.
+        The keys are indices of the CFA fragment dimensions,
+        e.g. ``(1, 0, 0 ,0)``.
 
         .. versionadded:: 3.14.0
 
         :Parameters:
 
-            var: `CFAPython.CFAVariable.CFAVariable`
-                The CFA aggregation variable.
+            copy: `bool`, optional
+                Whether or not to return a copy of the aggregation
+                dictionary. By default a deep copy is returned.
+
+                .. warning:: If False then changing the returned
+                             dictionary in-place will change the
+                             aggregation dictionary stored in the
+                             {{class}} instance, **as well as in any
+                             copies of it**.
+
+        :Returns:
+
+            `dict`
+                The aggregation data dictionary.
+
+        **Examples**
+
+        >>> a.shape
+        (12, 1, 73, 144)
+        >>> a.get_fragment_shape()
+        (2, 1, 1, 1)
+        >>> a.get_aggregated_data()
+        {(0, 0, 0, 0): {'file': 'January-June.nc',
+          'address': 'temp',
+          'format': 'nc',
+          'location': [(0, 6), (0, 1), (0, 73), (0, 144)]},
+         (1, 0, 0, 0): {'file': 'July-December.nc',
+          'address': 'temp',
+          'format': 'nc',
+          'location': [(6, 12), (0, 1), (0, 73), (0, 144)]}}
+
+        """
+        aggregated_data = self._get_component("aggregated_data")
+        if copy:
+            aggregated_data = deepcopy(aggregated_data)
 
-            frag_loc: `tuple` of `int`
-                The new key, that must be index of the CFA fragment
-                dimensions, e.g. ``(1, 0, 0, 0)``.
+        return aggregated_data
 
-            aggregated_data: `dict`
-                The aggregated data dictionary to be updated in-place.
+    def get_fragmented_dimensions(self):
+        """Get the positions dimension that have two or more fragments.
```

Comment:
```suggestion
        """Get the dimension positions that have two or more fragments.
```
reads/flows more naturally and in general is clearer, I think?

Reply:
Yes, indeed. Might go for "Get the positions of dimensions that have two or more fragments."